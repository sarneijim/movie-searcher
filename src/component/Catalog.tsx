import React, { Component } from "react";
import { GridList, GridListTile, GridListTileBar } from "@material-ui/core";
import withWidth, { isWidthUp } from "@material-ui/core/withWidth";
import { Breakpoint } from "@material-ui/core/styles/createBreakpoints";
import { Link } from "react-router-dom";
import styles from "../styles/catalog.module.css";
import { connect } from "react-redux";
import { compose } from "redux";
import { imgPath } from "../request";

class Catalog extends Component<any, any> {
  constructor(props: { catalog: IMedia[]; width: any }) {
    super(props);
    this.state = { width: props.width };
  }

  render() {
    const catalog: IMedia[] = this.props.catalog || [];
    const getCols = (screenWidth: Breakpoint) => {
      if (isWidthUp("lg", screenWidth)) {
        return 5;
      }
      if (isWidthUp("md", screenWidth)) {
        return 3;
      }
      return 2;
    };
    const cols = getCols(this.state.width);
    return (
      <div className={styles.root}>
        <GridList cellHeight={180} cols={cols}>
          {catalog.map((item: IMedia) => (
            <GridListTile key={item.id} className={styles.title}>
              <Link to={`/${item.media_type}/${item.id}`}>
                <img
                  className={styles.img}
                  src={`${imgPath}${
                    item.poster_path ? item.poster_path : item.profile_path
                  }`}
                  alt={item.title || item.name}
                />
                <GridListTileBar
                  title={item.title || item.name}
                  subtitle={<span>{item.overview}</span>}
                />
              </Link>
            </GridListTile>
          ))}
        </GridList>
      </div>
    );
  }
}
const mapStateToProps = (state: any) => {
  return {
    catalog: state.catalog,
  };
};
export default compose<any>(connect(mapStateToProps), withWidth())(Catalog);
