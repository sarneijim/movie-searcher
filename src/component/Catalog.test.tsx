import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import Catalog from "./Catalog";
import { BrowserRouter, Route } from "react-router-dom";
const mockStore = configureStore([]);

describe("Catalog", () => {
  let component: any;
  const store = mockStore({
    catalog: [
      { name: "abc", id: 1 },
      { name: "bcd", id: 2 },
    ],
  });

  beforeEach(() => {
    component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Catalog width="lg" />
        </Provider>
      </BrowserRouter>
    );
  });

  afterEach(() => {
    component = null;
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should match the snapshot", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
