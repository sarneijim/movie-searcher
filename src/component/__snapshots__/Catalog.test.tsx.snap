// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog should match the snapshot 1`] = `
<div
  className="root"
>
  <ul
    className="MuiGridList-root"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root title"
      style={
        Object {
          "height": 184,
          "padding": 2,
          "width": "20%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/undefined/1"
          onClick={[Function]}
        >
          <img
            alt="abc"
            className="img"
            src="https://image.tmdb.org/t/p/w500undefined"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                abc
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                <span />
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root title"
      style={
        Object {
          "height": 184,
          "padding": 2,
          "width": "20%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/undefined/2"
          onClick={[Function]}
        >
          <img
            alt="bcd"
            className="img"
            src="https://image.tmdb.org/t/p/w500undefined"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                bcd
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                <span />
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
  </ul>
</div>
`;
